proposals/0061-fair-congestion-control.md

```diff
@@ -0,0 +1,258 @@
+---
+simd: '0061'
+title: Fair congestion control with intra-block exponential local base fee
+authors:
+  - Ryo Onodera (Solana Labs)
+category: Standard/Meta
+type: Core
+status: Draft
+created: 2023-07-05
+feature: (fill in with feature tracking issues once accepted)
+---
+
+## Summary
+
+This feature is a fair congestion-control mechanism in the form of an extension
+to local fee markets while leaving their locality of transaction fee dynamics
+intact.
+
+To that end, it introduces a exponentially-scaled dynamic base fees to
+individual local fee markets. It also attains very short feedback loop of
+per-tx frequency to maintain full efficacy of Solana's peculiar execution
+model compared to other blockchains: multi-threaded and low latency.
+
+This is realized by means of some incentive tweaks to combat against the
+obvious base fee manipulation with such short intervals.
+
+## Motivation
+
+(still in half-finished...)
+
+- Write lock cu limit is bad (bot can lock out at the very first of block for
+  the entire duration of whole blocktime (400ms)
+- Increased Defi activities around any volatile financial markets could starve
+  payment transactions for extended time
+- Inter-block and linear Voluntary fee escalation with vanilla fee market
+  auction can't guarantee the scheduling deadline of casual payment txes (which
+  needs 99.99% sub-second confirmation at very minimum).
+
+## Overview of the Design
+
+This proposal tries to localize congestions by means of increasing minimum
+required `cu_price`s for each write-locked addresses (= `base_cu_price`), which
+is newly introduced by this proposal. This increase will be calculated
+exponentially, measured by the CU consumed by each addresses at the moment.
+This means a transaction must cost the sum of `requested_cu * base_cu_price`
+for all of its write-locked addresses at least. This results in selectively
```

Comment:
> is `base_cu_price` per writ-locked account

yes.

> will/should there be a `base-cu-price` for the block itself (in case the congestion is for block)?

no, there shouldn't be no such thing as written below in the proposal.